iree/turbine/kernel/wave/codegen/handlers.py

```diff
@@ -375,6 +376,44 @@ def handle_shuffle(emitter: WaveEmitter, node: fx.Node):
 
     emitter.bind_node_proxy(node, IRProxyValue(vec_result))
 
+def handle_atomic_op(op):
+    def decorator(binary_fn: Callable[[Value, Value], OpResult]):
+        @handle_op(op)
+        def handle_generic_atomic(emitter: WaveEmitter, node: fx.Node):
+            try:
+                lhs, rhs = node.args
+            except ValueError as e:
+                raise ValidationError("Malformed arguments") from e
+            lhs = cast_py_value(emitter, lhs)
+            rhs = cast_py_value(emitter, rhs)
+            lhs_data_type = get_type_or_element_type(lhs.ir_value.type)
+            rhs_data_type = get_type_or_element_type(rhs.ir_value.type)
+            if not MemRefType.isinstance(rhs.ir_value.type):
+                op = get_custom(node)
+                raise ValidationError(f"Expected rhs to be Memref type for\n"
+                    f"{op}\nGot\n"
+                    f"rhs: {rhs.ir_value.type}\n")
+            if lhs_data_type != rhs_data_type:
+                op = get_custom(node)
+                raise ValidationError(
+                    f"Expected lhs and rhs to have same data type for\n"
+                    f"{op}\nGot\n"
+                    f"lhs: {lhs_data_type} vs rhs: {rhs_data_type}\n"
+                )
+            if _is_float_type(lhs_data_type):
```

Comment:
Could you add a TODO above this with your learnings.

Reply:
done